dataset-metadata/metadata.ts

```diff
@@ -1139,3 +1139,16 @@ export const baseDatasetForReferenceGenome = (genome: ReferenceGenome): DatasetI
 
 export const hasJointFrequencyData = (datasetId: DatasetId): boolean =>
   getMetadata(datasetId, 'hasJointFrequencyData')
+
+export const getTopLevelDataset = (datasetId: DatasetId) =>
+  /* eslint-disable no-nested-ternary */
```

Comment:
this is hardly the most egregious nested ternary I've ever seen but still, let's go with this kind of style instead:

```
if(isV4(datasetId)) {
  return 'v4'
}

if(isV3(datasetId)) {
  return 'v3'
}
...
// no if needed here since we're at the default case
return 'default'
```

Multiple returns in a function are a little controversial but personally I think that's fine especially in a situation like this where all the different cases are going to be very similar in form. Some languages allow "pattern matching" of values, which is kind of a formalization of this style built into the language.